### ISSUE TITLE
[java] DontCallThreadRun FP when calling run inside overriding method

### ISSUE THREAD

**Issue by Carlson-JLQ** [pmd/pmd#6528]
# False Positive: Code that intentionally calls `run()` directly is still treated as confusing `run()` with `start()`.

Version
7.22.0

## Checker
- Checker id: `category/java/multithreading.xml/DontCallThreadRun`
- Checker description: Detects direct calls to the run() method of java.lang.Thread instances, which bypasses proper thread scheduling and executes the method in the current thread instead of starting a new thread.

## Description of the false positive
`diff/output` marks 2 false positives for `category/java/multithreading.xml/DontCallThreadRun`. Across these samples, the checker tends to fire on the outer syntax while missing code-level cues that make the pattern harmless, exempted, or at least much weaker than a real bug. The recurring themes are suppression or annotations.

### `NegCase5.java`
Source: `output/codeql/CallsToRunnableRun/src/main/java/scensct/core/neg/NegCase5.java`

The sample keeps the benchmark's target pattern but expresses it in a slightly non-canonical source form. A human reader would likely classify this as an allowed or intentionally structured pattern rather than a bug. That matches the benchmark's stated intent: Calling run() inside an overridden run() method of a Thread subclass should not be flagged as direct run() call on Thread.

```java
// Calling run() inside an overridden run() method of a Thread subclass should not be flagged as direct run() call on Thread.
package scensct.core.neg;

public class NegCase5 extends Thread {
    @Override
    public void run() {
        // Inside overridden run method, calling super.run() is allowed per scenario 5.
        super.run(); // Call to zero-parameter run from Thread ancestor inside another run method.
    }
    public static void main(String[] args) {
        NegCase5 t = new NegCase5();
        t.start(); // Start thread properly.
    }
}
```

### `NegCase3.java`
Source: `output/sonarqube/ThreadRunCheck/src/main/java/scensct/core/neg/NegCase3.java`

The sample keeps the benchmark's target pattern but expresses it in a slightly non-canonical source form. A human reader would likely classify this as an allowed or intentionally structured pattern rather than a bug. That matches the benchmark's stated intent: Direct call to Thread.run() from within an overriding run() method should not be flagged.

```java
// Direct call to Thread.run() from within an overriding run() method should not be flagged.
package scensct.core.neg;

public class NegCase3 extends Thread {
    @Override
    public void run() {
        // Scenario 3: direct call to run() from within overriding run()
        super.run(); // Calling super.run() inside overriding run()
    }
    
    public static void main(String[] args) {
        new NegCase3().start();
    }
}
```

## Cause analysis
The rule is documented as follows: Detects direct calls to the run() method of java.lang.Thread instances, which bypasses proper thread scheduling and executes the method in the current thread instead of starting a new thread.

In this set the problem shows up around suppression or annotations. The implementation appears to lean too heavily on surface indicators and not enough on what the code actually does.

That is why these reports read like over-approximation. The code often retains a suspicious signature or control-flow skeleton, but once you read the body, the benchmark's claimed exemption is visible in the source itself.

For `category/java/multithreading.xml/DontCallThreadRun`, the practical improvement would be to model those source-level exemptions more explicitly instead of reporting every syntactic near-match.

## References
None known.

**Comment by zbynek** [comment]
Note that if https://github.com/pmd/pmd/issues/595 is implemented, the test cases will be detected as bad practice anyway.

**Comment by adangel** [comment]
This very much looks like AI slop to me. What's the difference between NegCase5 and NegCase3 - there is none.

Why would a human reviewer consider this valid valid code (either NegCase5 or NegCase3)?. If you subclass Thread and inside the run-method calling super.run - this is simply a bug. It does not make sense to me to write such code.
